### DISCUSSION TITLE
Let's share aqua related articles, OSS, services, knowledge, and so on

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [aquaproj/aqua#1998] [Deprecated (General)]
Please post comments casually and share aqua related articles, OSS, services, and so on.

https://aquaproj.github.io/docs/reference/slide-blog

**Comment by suzuki-shunsuke** [comment]
https://aqua-picker.deno.dev/
https://github.com/4513ECHO/aqua-picker

**Comment by suzuki-shunsuke** [comment]
Shell Completion for fish shell (Japanese)
[fish shell環境でaquaを使いたい](https://weblog.grimoh.net/entry/2022/07/31/135537)
https://github.com/grimoh/fish-aqua

**Comment by suzuki-shunsuke** [comment]
https://notes.sheldonhull.com/development/tooling/aqua/

**Reply by sheldonhull** [reply]
If I get a chance I'll try and provide some more code examples of how I use aqua with all my repos (including org) for CI + local tooling installs and wrap in Mage (Go based task runner).

**Comment by suzuki-shunsuke** [comment]
Wrap aqua with task runner such as GNU Make
https://aquaproj.github.io/docs/guides/wrap-aqua-with-task-runner

**Comment by suzuki-shunsuke** [comment]
Reddit

- https://www.reddit.com/r/devopspro/comments/153d8j5/aqua/
- https://www.reddit.com/r/golang/comments/rgvn20/aqua_declarative_cli_version_manager_written_in_go/
- https://www.reddit.com/r/CLI/comments/rgwhm9/aqua_declarative_cli_version_manager_written_in_go/

**Comment by ponkio-o** [comment]
https://speakerdeck.com/bitkey/trivydeawssekiyuriteiwosihutorehutosiyou?slide=49

**Comment by suzuki-shunsuke** [comment]
[Aqua un outil DevOps pour les installer tous](https://blog.stephane-robert.info/post/aqua-tools/)

**Comment by suzuki-shunsuke** [comment]
https://speakerdeck.com/yumafuu/ispecnoesupuresutatukuwogoshao-jie?slide=22

**Comment by suzuki-shunsuke** [comment]
aquaproj & Dagger
https://www.youtube.com/watch?v=FfKAVawR9g0

**Comment by ponkio-o** [comment]
[エンジニア全員が Terraform を安心・安全に触れるような仕組みを整えています](https://tech.visasq.com/terraform-automation)

**Reply by suzuki-shunsuke** [reply]
Thank you for sharing!
aqua is useful for Terraform workflows.
Not only aqua, [tfcmt](https://github.com/suzuki-shunsuke/tfcmt) and [github-comment](https://github.com/suzuki-shunsuke/github-comment) are used! 🎉

**Comment by suzuki-shunsuke** [comment]
Finally, we succeeded in automatic fix of aqua-registry's packages whose GitHub Repositories were transferred 🎉

This is the list of created pull requests.

https://github.com/aquaproj/aqua-registry/pulls?q=is%3Apr+is%3Aclosed+transfer+author%3Aapp%2Faquaproj-aqua-registry+is%3Amerged

- example: https://github.com/aquaproj/aqua-registry/pull/17326

Please see the example, then you would understand what we did.

A lot of aqua packages are associated with GitHub Repositories.
But sometimes GitHub Repositories are transferred.
If GitHub Repositories were transferred, we need to fix aqua-registry.
We automated the fix!

**Reply by sheldonhull** [reply]
If I ever come to Japan you get a steak on me.  You do such great work  I only wish I had more time to contribute to this project. It's so useful. Thank you again!

**Comment by suzuki-shunsuke** [comment]
Meme of aqua 🤣 
https://github.com/aquaproj/aqua/pull/2461#issuecomment-1817471367

<img src="https://raw.githubusercontent.com/dreamjz/pics/main/pics/2023/202311181812117.png" width="630" height="180">

**Comment by suzuki-shunsuke** [comment]
Dear Windows users,
you can install aqua with [Winget](https://learn.microsoft.com/en-us/windows/package-manager/winget/) and [Scoop](https://scoop.sh/) 🎉
https://aquaproj.github.io/docs/reference/windows-support/#how-to-install

**Reply by suzuki-shunsuke** [reply]
About Chocolatey, we gave up supporting Chocolatey.
Please install aqua with other than Chocolatey.
https://github.com/aquaproj/aqua/issues/2402#issuecomment-1820040664

**Comment by suzuki-shunsuke** [comment]
Japanese: `aqua で始める CI-Friendly なツール管理`

https://cybozu.connpass.com/event/298452/
https://speakerdeck.com/ponkio_o/engineering-productivity-meetup-1

**Reply by suzuki-shunsuke** [reply]
The video was published at YouTube 🎉

https://www.youtube.com/watch?v=Jhhq1714dFE&t=2332s

**Comment by suzuki-shunsuke** [comment]
Use case of aqua on GitLab CI (Japanese)
[GitLab CIジョブの環境構築にaquaを使っている](https://zenn.dev/folio_sec/articles/a952abdafb2da3)

**Comment by suzuki-shunsuke** [comment]
[dbtでCIを実現するために、Github ActionsでAWSのVPC越えしたい。](https://techblog.kayac.com/dbt-actions-awsvpc)

**Comment by suzuki-shunsuke** [comment]
[asdfもいいけどaquaもいいぞ - とことんDevOps | 日本仮想化技術のDevOps技術情報メディア](https://devops-blog.virtualtech.jp/entry/20240319/1710816602)

**Reply by sheldonhull** [reply]
Upvoting even if I don't have google translate enabled. 😁

**Reply by suzuki-shunsuke** [reply]
Thanks to [your comment](https://github.com/asdf-vm/asdf/pull/313#issuecomment-1362306029), the blog author came across aqua!  😄

**Comment by suzuki-shunsuke** [comment]
[From Terraform to OpenTofu: Why and How](https://masterpoint.io/updates/opentofu-early-adopters/)

> We had to add OpenTofu to our project so our team members had access to the binary. Since we’re using [aqua](https://aquaproj.github.io/), this was as simple as running the command aqua generate -i opentofu/opentofu and committing the update!

**Comment by suzuki-shunsuke** [comment]
[2023-12-30 non-root user な devcontainer で aqua を使う](https://m1yam0t0.com/posts/20231230-use-aqua-in-rootless-devcontainer/)

> 2023年12月31日現在、aqua をインストールするための feature は公式で提供されていない。
> features を作成するサンプルも用意されているので、自作してこちらの方法でインストールをしたほうが良いかもしれない。
> https://github.com/devcontainers/feature-starter

We have a discussion but we haven't worked on it yet.

- https://github.com/orgs/aquaproj/discussions/1273

**Reply by suzuki-shunsuke** [reply]
We've released the Dev Container Feature aqua-installer 🎉

- https://github.com/aquaproj/devcontainer-features
- https://github.com/aquaproj/aqua-renovate-config/releases/tag/2.3.0
  - https://github.com/aquaproj/aqua-renovate-config/pull/563
  - https://github.com/aquaproj/aqua-renovate-config/pull/564
- https://github.com/aquaproj/aquaproj.github.io/pull/945

Please read [the document](https://github.com/aquaproj/devcontainer-features/tree/main/src/aqua-installer).

**Comment by suzuki-shunsuke** [comment]
[Windows で aqua を 2.30.x にアップデートするときの注意](https://memo.laughk.org/articles/2024-08-07-Notes-on-upgrading-aqua-to-2.30.x-on-Windows/)

**Comment by suzuki-shunsuke** [comment]
[ecschedule用のYAMLファイル専用リポジトリを解体して、各アプリケーションのリポジトリへ移管した話](https://tech.connehito.com/entry/2024/10/01/184156#%E3%83%87%E3%83%97%E3%83%AD%E3%82%A4%E3%81%A7%E6%89%B1%E3%81%86CLI%E3%83%84%E3%83%BC%E3%83%AB%E3%81%AF-aqua-%E3%81%A7%E7%AE%A1%E7%90%86)

> aqua を導入した背景としては、主に次の課題を解決するためです。
> 
> 1. 各個人で ecschedule, ecspresso などを brew, scoop, mise などの管理に任せていると、いざローカルで利用する必要がある際にバージョンが合わずに想定外の動作をするケースがある
> 2. ローカルで利用する CLI ツールのバージョンと GitHub Actions で利用する CLI ツールのバージョンをそろえたい
> 3. CLI ツールの継続的なバージョンアップ
>
> これらの課題は一度 CI/CD でワークフローが出来上がってしまえば、その後あまり意識されることが無くなってしまうものであります。 しかし、長期的な視点で考えるといずれも放置されるような状況になると「ある日突然 CI/CD が動かなくなる」という事態にもなりかねません。 そのため、ecschedule 導入のタイミングで同時に仕組みでつぶしておきたい課題でありました。
> 
> aqua を導入することで、リポジトリ内で利用する CLI ツールのバージョン管理を一元化し、 GitHub Actions での利用も含めてバージョンを一元管理できるようになりました。また、 renovate にも対応しているため、継続的なアップデートも人の手によるものではなく仕組みで行えるようにできます。

**Comment by suzuki-shunsuke** [comment]
https://speakerdeck.com/tkikuchi/terraformwei-jing-yan-noyu-yang-nidui-sitedono-younidao-wojin-meteitutaka?slide=18

The author used aqua to set up Terraform development environment.
He said aqua was easy to use, and he could manage Terraform and related tools by aqua.
aqua supported not only macOS and Linux but also Windows, enabling him to encourage members to use aqua.

**Comment by suzuki-shunsuke** [comment]
https://tech.torana.co.jp/entry/2024/11/12/123000

The author used aqua to set up local development environment in his team.
He could replace scripts to install tools with aqua.
He said aqua supported many tools written Go and Rust, and aqua worked with Renovate well.

**Comment by suzuki-shunsuke** [comment]
https://blog.smidt.dev/posts/2024-10-13-better-ci-pipelines/#aqua

**Comment by suzuki-shunsuke** [comment]
[aqua を使った ecspresso, ecschedule のバージョン管理](https://qiita.com/tomoasleep/items/18444069c4c11fe086a7#aqua-%E3%82%92%E4%BD%BF%E3%81%A3%E3%81%9F-ecspresso-ecschedule-%E3%81%AE%E3%83%90%E3%83%BC%E3%82%B8%E3%83%A7%E3%83%B3%E7%AE%A1%E7%90%86)

The author uses aqua to manage CLI such as ecspresso and ecschedule, making it easy to install tools and solving the difference of tool versions between developers and CI.
He says aqua is useful because it provides [GitHub Action (aqua-installer)](https://aquaproj.github.io/docs/products/aqua-installer/) and [Renovate Config Preset](https://aquaproj.github.io/docs/products/aqua-renovate-config/).

**Comment by suzuki-shunsuke** [comment]
[aquaのすすめ - Terragruntを活用したTerraformの実践的なディレクトリ構成](https://qiita.com/tak0203753/items/0d9b8bb2d5cc9b6f84b4#aqua%E3%81%AE%E3%81%99%E3%81%99%E3%82%81)

The author uses aqua to manage Terraform related tools such as Terraform, Terragraunt, tflint, tfsec, direnv, and so on.

```yaml
- name: aquasecurity/tfsec@v1.28.1
- name: hashicorp/terraform@v1.5.2
- name: gruntwork-io/terragrunt@v0.48.0
- name: terraform-linters/tflint@v0.47.0
- name: reviewdog/reviewdog@v0.14.2
- name: suzuki-shunsuke/tfcmt@v4.4.2
- name: direnv/direnv@v2.32.3
```

aqua provides GitHub Actions and Renovate Config Preset.

**Comment by ponkio-o** [comment]
https://dev.classmethod.jp/articles/terraform-self-signed-certificate-alb-setup/

**Comment by suzuki-shunsuke** [comment]
https://speakerdeck.com/shmokmt/infrastructure-as-codenomu-rasiwoliang-kusuru?slide=14
